types/types_test.go

```diff
@@ -26,21 +26,66 @@ func TestTransportPortConv(t *testing.T) {
 }
 
 func TestTransportPortBindingConv(t *testing.T) {
-	sform := "tcp/172.28.30.23:80/112.0.43.56:8001"
-	pb := &PortBinding{
-		Proto:    TCP,
-		IP:       net.IPv4(172, 28, 30, 23),
-		Port:     uint16(80),
-		HostIP:   net.IPv4(112, 0, 43, 56),
-		HostPort: uint16(8001),
+	input := []struct {
+		sform      string
+		pb         PortBinding
+		shouldFail bool
+	}{
+		{ // IPv4 -> IPv4
+			sform: "tcp/172.28.30.23:80/112.0.43.56:8001",
+			pb: PortBinding{
+				Proto:    TCP,
+				IP:       net.IPv4(172, 28, 30, 23),
+				Port:     uint16(80),
+				HostIP:   net.IPv4(112, 0, 43, 56),
+				HostPort: uint16(8001),
+			},
+		},
+		{ // IPv6 -> IPv4
+			sform: "tcp/[2001:db8::1]:80/112.0.43.56:8001",
+			pb: PortBinding{
+				Proto:    TCP,
+				IP:       net.IP{0x20, 0x01, 0x0d, 0xb8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1},
+				Port:     uint16(80),
+				HostIP:   net.IPv4(112, 0, 43, 56),
+				HostPort: uint16(8001),
+			},
+		},
+		{ // IPv4inIPv6 -> IPv4
+			sform: "tcp/[::ffff:172.28.30.23]:80/112.0.43.56:8001",
+			pb: PortBinding{
+				Proto:    TCP,
+				IP:       net.IPv4(172, 28, 30, 23),
+				Port:     uint16(80),
+				HostIP:   net.IPv4(112, 0, 43, 56),
+				HostPort: uint16(8001),
+			},
+		},
+		{ // IPv4 -> IPv4 zoned
+			sform:      "tcp/172.28.30.23:80/169.254.0.23%eth0:8001",
+			shouldFail: true,
+		},
+		{ // IPv4 -> IPv6 zoned
+			sform:      "tcp/172.28.30.23:80/[fe80::1ff:fe23:4567:890a%eth0]:8001",
+			shouldFail: true,
+		},
 	}
 
-	rc := new(PortBinding)
-	if err := rc.FromString(sform); err != nil {
-		t.Fatal(err)
-	}
-	if !pb.Equal(rc) {
-		t.Fatalf("FromString() method failed")
+	for _, in := range input {
+		rc := new(PortBinding)
+		err := rc.FromString(in.sform)
+		if in.shouldFail {
+			if err == nil {
```

Comment:
you can use the require package `require.NoError(t, err)` and `require.Error(t, err)`
it's more compact

Reply:
Thanks.   I've seen that used elsewhere but I was following the existing style in this file.

Do we have a coding style guide?   Should we be using testify for all new tests (and converting existing ones over as we update them)?

Reply:
I was suggesting mainly for brevity and readability. Don't think is worth going back and update all the tests, maybe if we spend time updating some then we can simplify it, but not mandatory